src/main/java/org/apache/cassandra/db/ColumnFamilyStore.java

```diff
@@ -519,8 +519,8 @@ public CompositeData finishLocalSampling(String samplerType, int count) throws O
         JsonArray counters = tableSamplerResult.getJsonArray((samplerType.equalsIgnoreCase("reads")) ? "read" : "write");
         long size = 0;
         if (counters != null) {
-            size = counters.size();
-            for (int i = 0; i < counters.size(); i++) {
+            size = (count > counters.size()) ? counters.size() : count;
```

Comment:
std::min() would have been clearer, but I'm fine with this version too. I'll commit.